plone/app/discussion/conversation.py

```diff
@@ -89,8 +89,22 @@ def enabled(self):
 
     @property
     def total_comments(self):
+        return self.public_comments()
+
+    def public_comments(self):
+        """Count total comments.
+
+        The @property decorator used on ``total_comments`` method doesn't work
+        when combined with acquisition, that's a fact of life.
+
+        Thus the total_comments indexer (on p.a.discussion.catalog.py) fails
+        to index properly if acquisition is involved.
+
+        To solve it, this method takes over the indexing and is being used by
```

Comment:
Well, any usage of p.a.discussion like:

```python
conversation = IConversation(self.context)
return conversation.total_comments
```

Will need to be replaced for:

```python
conversation = IConversation(self.context)
return conversation.public_comments()
```

Or if we do replace it, parenthesis need to be appended to ``total_comments``.

On our freitag.de packages for example that means to at least 4 to 5 lines having to be changed. That's completely fine for me, I don't know what Plone's policy is for backwards compatibility at method level on a stable (4.3.x version) release.

For 5.0 I would say that yes, let's change it. For 4.3.4 (as this PR is about) maybe is not such a good idea to break backwards compatibility on a micro release (what will end up being 4.3.5).